MediaBrowser.Providers/MediaInfo/AudioFileProber.cs

```diff
@@ -172,9 +172,9 @@ private async Task FetchDataFromTags(Audio audio, Model.MediaInfo.MediaInfo medi
 
             track.Title = string.IsNullOrEmpty(track.Title) ? mediaInfo.Name : track.Title;
             track.Album = string.IsNullOrEmpty(track.Album) ? mediaInfo.Album : track.Album;
-            track.Year ??= mediaInfo.ProductionYear;
-            track.TrackNumber ??= mediaInfo.IndexNumber;
-            track.DiscNumber ??= mediaInfo.ParentIndexNumber;
+            track.Year = track.Year is null or 0 ? mediaInfo.ProductionYear : track.Year;
+            track.TrackNumber = track.TrackNumber is null or 0 ? mediaInfo.IndexNumber : track.TrackNumber;
+            track.DiscNumber = track.DiscNumber is null or 0 ? mediaInfo.ParentIndexNumber : track.DiscNumber;
```

Comment:
We usually don't use inline-linq like `or` in the code

Reply:
But in this case it is much cleaner than having nested assignments.